crates/backend/src/ast.rs

```diff
@@ -1072,3 +1105,24 @@ fn replace_self(name: &Ident, item: &mut syn::ImplItem) {
 
     syn::visit_mut::VisitMut::visit_impl_item_mut(&mut Walk(name), item);
 }
+/// Extract the documentation comments from a Vec of attributes
+fn extract_doc_comments(attrs: &Vec<syn::Attribute>) -> Vec<String> {
+    attrs
+    .iter()
+    .filter_map(|a| {
+        //We only care about outer comments for now
+        if a.style == AttrStyle::Outer {
```

Comment:
Shouldn't this also check that the attributes are `doc`? It seems to me that this will collect non-`doc` attributes as well, which we don't want.

Reply:
Good catch, I am pretty new at using `syn`/`proc-macro2`, I updated this to check the path segments for a "doc" `Ident` instead of just capturing outer attributes. Let me know if you know of a better way to differentiate doc comment attributes.